pysd/tools/ncfiles.py

```diff
@@ -0,0 +1,416 @@
+        Convert a dict to a pandas Dataframe.
+
+        Parameters
+        ----------
+        d: dict
+            Dictionary to convert to pandas DataFrame.
+
+        """
+        if "time" not in d:
+            raise KeyError("Missing time key.")
+
+        return pd.DataFrame(d).set_index('time')
+
+    @staticmethod
+    def df_to_csv(df: pd.DataFrame, outfile: Path,
+                  time_in_row: Optional[bool]=False
+                  ) -> None:
+        """
+        Store pandas DataFrame into csv or tab file.
+
+        Parameters
+        ----------
+        df: pandas.DataFrame
+            DataFrame to save as csv or tab file.
+        outfile: str or pathlib.Path
+            Path of the output file.
+        time_in_row: bool
+            Whether time increases along a column or a row.
+
+        Returns
+        -------
+        None
+        """
+        if isinstance(outfile, str):
+            outfile = Path(outfile)
+
+        out_fmt = outfile.suffix
+
+        if out_fmt not in NCFile.valid_export_file_types:
+            raise TypeError("Invalid output file format {out_fmt}\n"
+                            "Supported formats are csv and tab.")
+
+        outfile.parent. mkdir(parents=True, exist_ok=True)
+
+        if not isinstance(time_in_row, bool):
+            raise ValueError("time_in_row argument takes boolen values.")
+
+        # process output file path
+        if outfile.suffix == ".csv":
+            sep = ","
+            df.columns = [col.replace(",", ";") for col in df.columns]
+        else:
+            sep = "\t"
+
+        if time_in_row:
+            df = df.transpose()
+
+        # QUOTE_NONE used to print the csv/tab files as vensim does with
+        # special characterse, e.g.: "my-var"[Dimension]
+        df.to_csv(outfile, sep=sep, index_label="Time", quoting=QUOTE_NONE)
+
+        print(f"Data saved in '{outfile}'")
+
+    @staticmethod
+    def ds_to_df(ds: xr.Dataset,
+                 subset: Optional[list]=None,
+                 parallel: Optional[bool]=False,
+                 index_dim: Optional[str]="time"
+                 ) -> pd.DataFrame:
+        """
+        Convert xarray.Dataset into a pandas DataFrame.
+
+        Parameters
+        ----------
+        ds: xarray.Dataset
+            Dataset object.
+        subset: list
+            List of variables to export from the Dataset.
+        parallel: bool
+            When True, DataArrays are processed in parallel using dask delayed.
+            Setting parallel=True is highly recommended when DataArrays are
+            large and multidimensional.
+        index_dim: str
+            Name of dimensions to use as index of the resulting DataFrame
+            (usually "time").
+
+        Returns
+        -------
+        df: pandas.DataFrame
+            Dataframe with all colums specified in subset.
+
+        """
+        subset = NCFile._validate_ds_subset(ds, subset)
+
+        savedict = {}
+
+        if parallel:
```

Comment:
@enekomartinmartinez I am repeating a bit of code here. Do you think it's worth to move it into a separate function that takes one function or the other?